DigitalLearningSolutions.Web/Controllers/FrameworksController/Frameworks.cs

```diff
@@ -157,7 +157,7 @@ public IActionResult CreateNewFramework(string actionname, int frameworkId = 0)
             {
                 if (TempData[MultiPageFormDataFeature.AddNewFramework.TempDataKey] == null)

```

Comment:
Could we replace this repeated redirect code with a ServiceFilter implementation similar to that used in `DigitalLearningSolutions.Web\Controllers\Register\RegisterController.cs` before all GET methods in the multi page form, we have `[ServiceFilter(typeof(RedirectEmptySessionData<DelegateRegistrationData>))]`?

A variation of this called Redirect**ToError**EmptySessionData would allow us to decorate get methods with this service filter rather than repeat code. 

It occurred to me that we could use the existing service filter but to my mind, we don't want to redirect users to the start of the form in most cases and we don't always use the pattern of an `index.cshtml` view to start all of these flows (though perhaps we should), which that filter method is currently reliant on.

Reply:
I notice there is another ServiceFilter that might be of interest to look at: `DigitalLearningSolutions.Web\ServiceFilter\RedirectMissingMultiPageFormData.cs`

Reply:
Yes that's a good call re picking it up everywhere automatically in the GET method pre-processing. Will implement.

Reply:
@kevwhitt-hee I've implemented this as a new service filter throughout the frameworks controller (I couldn't find anywhere else where the code usually raises a 404(/410) if TempData is missing).